src/main/scala/com/wavesplatform/network/DiscardingHandler.scala

```diff
@@ -0,0 +1,32 @@
+package com.wavesplatform.network
+
+import com.wavesplatform.network.TransactionalMessagesRepo.TransactionMessageSpec
+import io.netty.channel.ChannelHandler.Sharable
+import io.netty.channel.{ChannelDuplexHandler, ChannelHandlerContext, ChannelPromise}
+import scorex.utils.ScorexLogging
+
+@Sharable
+class DiscardingHandler extends ChannelDuplexHandler with ScorexLogging {
+  @volatile
+  private var blockchainIsOutdated = true
+
+  override def channelRead(ctx: ChannelHandlerContext, msg: AnyRef) = msg match {
+    case RawBytes(code, _) if code == TransactionMessageSpec.messageCode && blockchainIsOutdated =>
+      log.trace(s"${id(ctx)} Discarding incoming message $code")
+    case _ => super.channelRead(ctx, msg)
+  }
+
+  override def write(ctx: ChannelHandlerContext, msg: scala.Any, promise: ChannelPromise) = msg match {
+    case BlockchainExpired =>
```

Comment:
I do not recall  we have such network messages.
What I see here is a message passing interface between application components.
If we have to do this, literally ANY other way is better -- actors, services, plain event bus(in comparison to non-type-safe-network-library-driven one)